scripts/generate_doc_stubs.py

```diff
@@ -0,0 +1,590 @@
+            content += "    options:\n"
+            content += "        heading_level: 1\n"
+
+            docs_path = f"clients/{self.service_name}/structures/{struct.name}.md"
+            with mkdocs_gen_files.open(docs_path, "w") as f:
+                f.write(content)
+
+        logger.info(f"Generated {len(structures)} structure files")
+
+    def _generate_unions(self, unions: list[UnionInfo]) -> None:
+        """Generate union documentation files."""
+        for union in unions:
+            content = f"::: {union.module_path}\n"
+            content += "    options:\n"
+            content += "        heading_level: 1\n\n"
+
+            # Add union members
+            if union.members:
+                content += "## Union Members\n\n"
+                for member in union.members:
+                    content += f"::: {member.module_path}\n"
+                    content += "    options:\n"
+                    content += "        heading_level: 3\n\n"
+
+            docs_path = f"clients/{self.service_name}/unions/{union.name}.md"
+            with mkdocs_gen_files.open(docs_path, "w") as f:
+                f.write(content)
+
+        logger.info(f"Generated {len(unions)} union files")
+
+    def _generate_enums(self, enums: list[EnumInfo]) -> None:
+        """Generate enum documentation files."""
+        for enum in enums:
+            content = f"::: {enum.module_path}\n"
+            content += "    options:\n"
+            content += "        heading_level: 1\n"
+            content += "        members: true\n"
+
+            docs_path = f"clients/{self.service_name}/enums/{enum.name}.md"
+            with mkdocs_gen_files.open(docs_path, "w") as f:
+                f.write(content)
+
+        logger.info(f"Generated {len(enums)} enum files")
+
+    def _generate_errors(self, errors: list[ErrorInfo]) -> None:
+        """Generate error documentation files."""
+        for error in errors:
+            content = f"::: {error.module_path}\n"
+            content += "    options:\n"
+            content += "        heading_level: 1\n"
+            content += "        members: true\n"
+
+            docs_path = f"clients/{self.service_name}/errors/{error.name}.md"
+            with mkdocs_gen_files.open(docs_path, "w") as f:
+                f.write(content)
+
+        logger.info(f"Generated {len(errors)} error files")
+
+
+def extract_service_name(package_name: str) -> str:
+    """Extract service name from client package name."""
+    return (
+        package_name
+        .replace("aws-sdk-", "")
+        .replace("-", " ")
+        .title()
+    )
+
+
+def main() -> int:
+    """Main entry point for the documentation generator."""
+    repo_root = Path(__file__).parent.parent.absolute()
+    output_dir = repo_root / "docs" / "clients"
+    clients_dir = repo_root / "clients"
+
+    try:
+        for client_dir in clients_dir.iterdir():
+            if client_dir.is_dir() and client_dir.name != "aws-sdk-python":
```

Comment:
When testing, this was picking up `.ruff_cache`. We should add another check that verifies there is a `pyproject.toml` file or something